Source/AwsCommonRuntimeKit/mqtt/Mqtt5Types.swift

```diff
@@ -625,26 +623,25 @@ public class LifecycleConnectSuccessData {
 }
 
 /// Defines signature of the Lifecycle Event Connection Success callback
-public typealias OnLifecycleEventConnectionSuccess = (LifecycleConnectSuccessData) -> Void
+public typealias OnLifecycleEventConnectionSuccess = (LifecycleConnectionSuccessData) -> Void
 
 /// Dataclass containing results of a Connect Failure Lifecycle Event.
-public class LifecycleConnectFailureData {
+public class LifecycleConnectionFailureData {
 
     /// Error which caused connection failure.
     public let crtError: CRTError
 
     /// Data model of an `MQTT5 CONNACK <https://docs.oasis-open.org/mqtt/mqtt/v5.0/os/mqtt-v5.0-os.html#_Toc3901074>`_ packet.
-    public let connackPacket: ConnackPacket
+    public let connackPacket: ConnackPacket?
```

Comment:
Was it an invalid case that connackPacket is null?

Reply:
Not sure what you mean by invalid case. 

ConnackPacket should be optional here since it's possible for the native client to emit a connection failure lifecycle event that both contains a failed connack (connect was sent and a negative connack received from iot broker) and a nil connack (e.g. connect timed out).